pkg/cluster/serviceprincipal.go

```diff
@@ -5,38 +5,34 @@ package cluster
 
 import (
 	"context"
-	"strings"
+	"fmt"
 	"time"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
-	azgraphrbac "github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
-	"github.com/jongio/azidext/go/azidext"
 	"k8s.io/apimachinery/pkg/util/wait"
 
 	"github.com/Azure/ARO-RP/pkg/api"
-	"github.com/Azure/ARO-RP/pkg/util/azureclient/graphrbac"
+	utilgraph "github.com/Azure/ARO-RP/pkg/util/graph"
 )
 
 // initializeClusterSPClients initialized clients, based on cluster service principal
 func (m *manager) initializeClusterSPClients(ctx context.Context) error {
 	spp := m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile
 	options := m.env.Environment().ClientSecretCredentialOptions()
-	credential, err := azidentity.NewClientSecretCredential(
+	tokenCredential, err := azidentity.NewClientSecretCredential(
 		m.subscriptionDoc.Subscription.Properties.TenantID,
 		spp.ClientID, string(spp.ClientSecret), options)
 	if err != nil {
 		return err
 	}
 
-	scopes := []string{m.env.Environment().ActiveDirectoryGraphScope}
-	spGraphAuthorizer := azidext.NewTokenCredentialAdapter(credential, scopes)
+	m.spGraphClient, err = m.env.Environment().NewGraphServiceClient(tokenCredential)
 
-	m.spApplications = graphrbac.NewApplicationsClient(m.env.Environment(), m.subscriptionDoc.Subscription.Properties.TenantID, spGraphAuthorizer)
-	return nil
+	return err
 }
 
 func (m *manager) clusterSPObjectID(ctx context.Context) error {
```

Comment:
Anywhere this is called, we can remove the refreshing action from now and test the previous theory about Authorization refreshing in azcore.  

Also on the fixClusterSPObjectID call too.

Reply:
What azcore refreshing theory? If I got it right, we reduce the amount of code here, which is always good direction.